src/liballoc_system/lib.rs

```diff
@@ -71,7 +72,49 @@ pub extern "C" fn __rust_usable_size(size: usize, align: usize) -> usize {
     imp::usable_size(size, align)
 }
 
-#[cfg(unix)]
+#[cfg(target_os = "redox")]
+mod imp {
```

Comment:
Just so we have a stub for potentially using a `libc` allocator, we implement `alloc_system` similarly to how the `unix` implementation works. If desired, these two imp blocks could be combined.